Native/NativeInputEventManager.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using UnityEngine.InputNew;
+using UnityEngineInternal.Input;
+
+namespace UnityEngine.Experimental.Input
+{
+	// Listens for native events and converts them into managed InputEvent instances.
+	class NativeInputEventManager : MonoBehaviour
+	{
+		void Start()
+		{
+			NativeInputSystem.onEvents += OnEvent;
+        }
+
+		internal void OnEvent(int eventCount, IntPtr eventData)
+		{
+			var currentDataPtr = eventData;
+			for (var i = 0; i < eventCount; i++)
+			{
+				unsafe
```

Comment:
is using unsafe code the only way to do this ?  iirc unsafe code has to be enabled at the project level and is generally not recommended

Reply:
This is so temporary enough that any answer that involves more than a few hours of dev time is not worth it

Reply:
Agreed. And as far as I know, unsafe code is the only way to get any use of out of a NativeEventCallback since we need to do pointer arithmetic to iterate through the events.